### PR TITLE
Refresh K8s Service Account Token

### PR DESCRIPTION
**Pull request by josunect**
k8s service token is never refreshed, so this PR is adding some logic to re-read the token:

- No more than once every minute
- If the SA token file has been modified

The auth strategy "anonymous" uses this token also as the user auth token. As the user is anonymous, get the SA to inherit its permissions. 

TODO: Still need to verify if the token is properly refreshed in the cache to be used when other auth strategies are set. 

Fixes https://github.com/kiali/kiali/issues/5070

### CODE REVIEW

File: kubernetes/token.go [hunk 1]
```diff
@@ -1,25 +1,91 @@
...
+
+	"github.com/kiali/kiali/log"
+)
 
 // Be careful with how you use this token. This is the Kiali Service Account token, not the user token.
 // We need the Service Account token to access third-party in-cluster services (e.g. Grafana).
 
-const DefaultServiceAccountPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+var DefaultServiceAccountPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+var refreshTime = int64(60)
```
Comment (Xunzhuo) [inline kubernetes/token.go:15]: Need we to make this configurable, since the refreshTime is different in different platforms?

**Review by Xunzhuo — COMMENTED** [review]
I know this is WIP, just a side comment.

File: kubernetes/token.go (hunk 1 above)
Comment (josunect) [reply kubernetes/token.go:15]: Thanks for your comment, @Xunzhuo 
I'm reviewing this.

**Comment by josunect** [comment]
I'm closing this PR as I created another different approach in https://github.com/kiali/kiali/pull/5386 
Setting expire at in the k8s configuration.

**Pull request closed without merging**